coordinator/src/reader/mod.rs

```diff
@@ -195,6 +195,26 @@ impl<O: VnodeOpener> Stream for CheckedCoordinatorRecordBatchStream<O> {
     }
 }
 
+pub fn vnode_broken_err(msg: String, code: &str) -> bool {
+    let e = tskv::error::Error::ReadTsm {
+        source: tskv::tsm::ReadTsmError::CrcCheck,
+    };
+
+    if models::error_code::ErrorCode::code(&e) == code {
+        return true;
+    }
+
+    if msg.contains("File Not Found") {
```

Comment:
This code is too difficult to maintain. A better solution？

Reply:
The best way is to keep throwing up each error code without any conversion; At the same time, future development should also follow this rule. But this change is significant

Reply:
I agree with you, and we should keep it on the way, moved this issue https://github.com/cnosdb/cnosdb/issues/1192  to v2.4.1